tools/c7n_gcp/c7n_gcp/actions/core.py

```diff
@@ -70,7 +70,25 @@ def process_resource_set(self, client, model, resources):
         for resource in resources:
             op_name = self.get_operation_name(model, resource)
             params = self.get_resource_params(model, resource)
-            result = self.invoke_api(client, op_name, params)
+            try:
+                result = self.invoke_api(client, op_name, params)
+            except HttpError as e:
+                if 'fingerprint' in e.reason:
+                    try:
+                        if model.refetch_fingerprint:
+                            resource = model.refetch_fingerprint(params, client, model, resource)
+                            params['body']['labelFingerprint'] = resource['labelFingerprint']
+                            result = self.invoke_api(client, op_name, params)
+                        else:
+                            raise
+                    except HttpError as e:
+                        if e.resp.status in self.ignore_error_codes:
+                            return e
+                        raise
+                else:
+                    if e.resp.status in self.ignore_error_codes:  # noqa
```

Comment:
re why noqa, ruff was complaining about e not being defined without this, flake8 was happy, so a bug in ruff afaics.